common/src/main/java/com/microsoft/identity/common/internal/cache/BrokerOAuth2TokenCache.java

```diff
@@ -166,11 +194,49 @@ public ICacheRecord save(@NonNull final GenericOAuth2Strategy oAuth2Strategy,
 
         final OAuth2TokenCache targetCache = isFoci ? mFociCache : mAppUidCache;
 
-        return targetCache.save(
+        final ICacheRecord result = targetCache.save(
                 oAuth2Strategy,
                 request,
                 response
         );
+
+        updateApplicationMetadataCache(
+                result.getIdToken().getClientId(),
+                result.getIdToken().getEnvironment(),
+                result.getRefreshToken().getFamilyId(),
+                mCallingAppUid
```

Comment:
Let's add the word "Process" to ensure that future developer understand that this is the process identifier as provided by the OS.